src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -295,6 +315,24 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
         continue;
       }
 
+      if (sched_cls_info.running_tasks.size() >= sched_cls_info.capacity) {
+        if (get_time_() > sched_cls_info.next_update_time) {
+          // Don't increase the capacity on the very first update since that
+          // will unblock a new task immediately, instead of after the timeout.
+          if (sched_cls_info.num_updates > 0) {
+            sched_cls_info.capacity++;
```

Comment:
Can we keep the capacity constant and just use the timer to decide whether to dispatch the task? It would be nice to make sure that the capacity eventually goes back down once other tasks finish.